### ISSUE TITLE
Deleted files not removed when keep_history: true

### ISSUE THREAD

**Issue by ysndr** [crazy-max/ghaction-github-pages#153]
### Behaviour

With `keep_history: true`, deleting a folder does not push file deletions to the remote repository.

#### Steps to reproduce this issue

Configure action with: `keep_history: true`

1. Push to gh_pages
    ```
    /a/file.txt
    /b/file.txt
    ```
2. Remove files/directory from parent repository
    ```
       /a/file.txt
     - /b/file.txt
    ```
3. Push to gh_pages

#### Expected behaviour

gh_pages repo/branch does not contain directory `b/`

#### Actual behaviour

Directory `b/` still present

### Configuration

* Repository URL (if public): https://github.com/ysndr/blog-test
* Build URL (if public): https://github.com/ysndr/beta (now configured with `keep_history: false`)

```yml
name: "Cleanup Preview"

on:
  pull_request:
    types: [ closed ]

env:
  GITHUB_PR_NUMBER: "${{github.event.number}}"

jobs:
  cleanup:
    runs-on: ubuntu-latest
    steps:
    - name: Pull release
      uses: actions/checkout@v2
      with:
        repository: ysndr/beta
        ref: gh-pages
        path: preview

    - name: Remove Preview
      run: |
        ls -la
        [ -d preview/${{ env.GITHUB_PR_NUMBER }} ] && rm -r preview/${{ env.GITHUB_PR_NUMBER }}
        ls -la

        rm  -rf preview/.git


    - name: Push
      uses: crazy-max/ghaction-github-pages@v2
      with:
        repo: ysndr/beta
        target_branch: gh-pages
        build_dir: ./preview
        committer: github-ci <notifications+ci-blog@ysndr.de>
        verbose: true
       keep_history: true
      env:
        GH_PAT: ${{ secrets.DEPLOY_TOKEN }}
```

### Logs
[logs_65.zip](https://github.com/crazy-max/ghaction-github-pages/files/6646636/logs_65.zip)

**Comment by crazy-max** [comment]
> Directory `b/` still present

You can't push an empty directory: https://git.wiki.kernel.org/index.php/GitFaq#Can_I_add_empty_directories.3F

**Comment by Barabas5532** [comment]
I have found the same issue. We do not want to push an empty directory. We want the deleted directory and all of its contents to be deleted from the `gh-pages` branch when the action runs.

I'm hosting the output of `strictdoc`, `doxygen` and other similar documentation tools in GitHub Pages. I started by setting up just `strictdoc` into the root of the github pages branch. Later I moved everything to a subdirectory.

The bug is that the files in the root of the repository still exist after I have moved them to a subdirectory. What I was uploading has a structure like this:

```
./html:
docs
requirements
```

However, the actual contents of the repository after the upload is like this:

![image](https://user-images.githubusercontent.com/28671465/183254358-56ddd68c-0625-4115-b198-2bc1763294c1.png)

The new folders `docs` and `requirements` were uploaded correctly, but files from a previous deployment (`_static`, `index.html`) have been left over. These files should have been deleted.

I think a workaround for this would be to `rm -r` all the files after the git clone step in the action. Git would see any identical regenerated files as not changed, and it would also see the deleted files correctly.
